src/components/windows/types/user/editUserDetails.jsx

```diff
@@ -0,0 +1,179 @@
+import { useForm } from "react-hook-form";
+import { CardContainer, CardContainerIcon, CardContainerInnerIcon, CardContainerInput, CardContainerInputWrapper, CardContainerSelectInput, CardContainerText, InnerContainer, InnerContainerRow, InnerContainerTitle, InputElement, InputLabel, PanelButton } from "../../../dashboard"
+import { useContext, useEffect, useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faAddressCard, faCalendar, faEnvelope, faUser } from "@fortawesome/free-regular-svg-icons";
+import { faMapPin, faPhone } from "@fortawesome/free-solid-svg-icons";
+import { User } from "@phoenixlan/phoenix.js";
+import { WindowManagerContext } from "../../windowManager";
+import { ErrorContainer } from "../../build";
+import { Notice } from "../../../containers/notice";
+
+export const EditUserDetails = ({ entries }) => {
+
+	// Inherit WindowManagerContext
+    const windowManager = useContext(WindowManagerContext);
+
+    const { register, handleSubmit, formState: { errors } } = useForm();
+
+	// Variables to keep track of changes in input components
+	const [ firstname, setFirstname ]			= useState(entries.firstname);
+	const [ lastname, setLastname ]				= useState(entries.lastname);
+	const [ username, setUsername ]				= useState(entries.username);
+	const [ email, setEmail ]					= useState(entries.email);
+	const [ phone, setPhone ]					= useState(entries.phone);
+	const [ guardianPhone, setGuardianPhone ]	= useState(entries.guardian_phone);
+	const [ address, setAddress ]				= useState(entries.address);
+	const [ postalCode, setPostalCode ]			= useState(entries.postal_code);
+	const [ birthdate, setBirthdate ]			= useState(entries.birthdate);
+	const [ gender, setGender ]					= useState(entries.gender);
+
+	// Error handling from API and JS
+	const [ error, setError ]					= useState(undefined);
+
+    const onSubmit = async (data) => {
+		try { 
+			await User.modifyUser(data.uuid, data.firstname, data.lastname, data.username, data.email, data.phone, data.guardian_phone, data.address, data.postal_code, data.birthdate, data.gender);
+			windowManager.exitWindow();
```

Comment:
This windowmanager stuff worries me, are these all popups or something? If so it's a very 2000s way of doing things, windows are a hassle to navigate. Better to just change the state of the page to edit boxes, or take the user to a new page with a new URL. That way you also don't need a back button :D

Reply:
Dont worry, we are not using the windowmanager anymore!